src/lib/resolver.js

```diff
@@ -15,53 +15,21 @@ var PDI = '\u2069';
 const meta = new WeakMap();
 const resolutionChain = new WeakSet();
 
-export function createEntry(node, lang) {
-  var keys = Object.keys(node);
+export function createEntry(id, entity, lang) {
+  meta.set(entity, {id, lang});
```

Comment:
A way to check that, other than `typeof === 'string'`, would be to not pass `id` directly into `createEntry`, and instead check if `entity.id` is present.  Simple strings don't have the `id` defined on them.